packages/web-components/src/components/content-group/content-group-heading.ts

```diff
@@ -28,12 +29,22 @@ class DDSContentGroupHeading extends StableSelectorMixin(LitElement) {
   @property({ reflect: true })
   slot = 'heading';
 
-  connectedCallback() {
-    super.connectedCallback();
+  /**
+   * Render the heading tag into the light DOM of this component.
+   */
+  protected _renderHeading() {
+    const template = document.createElement('template');
+    template.innerHTML = `<h3>${this.innerHTML.trim()}</h3>`;
+    const heading = template.content.firstChild;
+    render(html`${heading}`, this);
+  }
+
+  firstUpdated() {
+    this._renderHeading();
   }
 
   render() {
-    return html` <h3><slot></slot></h3>`;
+    return html`<slot></slot>`;
```

Comment:
Note that I went with a different approach from `<dds-content-block-heading>` b/c in that component, it strips the HTML of the slotted markup. Here we keep the spirit of allowing any markup to be slotted, b/c in some instances, AEM slots an `<a name="my-name"></a>` tag along side the text node heading.